posthog/kafka_client/client.py

```diff
@@ -87,12 +93,28 @@ def __init__(self, test=TEST):
             self.producer = helper.get_kafka_producer(retries=KAFKA_PRODUCER_RETRIES, value_serializer=lambda d: d)
         else:
             self.producer = KP(
-                retries=KAFKA_PRODUCER_RETRIES,
-                bootstrap_servers=KAFKA_HOSTS,
-                security_protocol=KAFKA_SECURITY_PROTOCOL or _KafkaSecurityProtocol.PLAINTEXT,
-                **_sasl_params(),
+                {
+                    "retries": KAFKA_PRODUCER_RETRIES,
+                    "bootstrap.servers": KAFKA_HOSTS,
+                    "security.protocol": KAFKA_SECURITY_PROTOCOL or _KafkaSecurityProtocol.PLAINTEXT.value,
+                    "statistics.interval.ms": 1000,
+                    "stats_cb": producer_metric_manager.send,
+                    **_sasl_params(),
+                }
             )
 
+        self._cancelled = False
+        self._poll_thread = Thread(target=self._poll_loop)
+        self._poll_thread.start()
+
+    def _poll_loop(self):
+        while not self._cancelled:
+            self.producer.poll(0.1)
```

Comment:
I was a bit skeptical about this but it's the same setting used by the [asyncio example](https://github.com/confluentinc/confluent-kafka-python/blob/master/examples/asyncio_example.py#L47) from upstream